libs/wingc/src/lifting.rs

```diff
@@ -281,8 +282,28 @@ impl<'a> Visit<'a> for LiftVisitor<'a> {
 		// CAPTURE
 		if self.should_capture_type(&node) {
 			// jsify the type so we can get the preflight code
-			let code = self.jsify_udt(&node);
-
+			let current_env = self.ctx.current_env().expect("an env");
+			let code = if let LookupResult::Found(SymbolKind::Namespace(root_namespace), _) =
+				current_env.lookup_ext(&node.root, None)
+			{
+				match &root_namespace.kind {
+					// types in wing files and the built-in wingsdk module already implement a helper to lift types
+					NamespaceKind::FileModule => self.jsify_udt(&node),
+					NamespaceKind::JSII { fqn } if fqn == WINGSDK_ASSEMBLY_NAME => self.jsify_udt(&node),
+
+					NamespaceKind::JSII { fqn } => format!(
+						"{{ _toInflightType: () => `require(\"{fqn}\").{}` }}",
+						node
+							.fields
+							.iter()
+							.map(|f| f.name.clone())
+							.collect::<Vec<String>>()
+							.join(".")
+					),
+				}
```

Comment:
I'd really like us to find a way to base the "lifting" logic on the type and not comparing names. Then implement this generically in the SDK. Maybe when the JSifier outputs the call to `_toInflightType` is should call some SDK function instead `toLifterType` that'll check if this object inherits `std.Resouce` and in that case just call its `_toInflightType` and if not inline the `require(...)` above.

Reply:
I wasn't sure what you mean by "comparing names". Like checking for the WINGSDK_ASSEMBLY_NAME? That's avoidable.

In any case, I've refactored to put more logic into the SDK. Unfortunately it doesn't look much better because SDK needs a lot of information passed to it. Lemme know what you think about it.

Reply:
By "comparing names" I meant checking for `WINGSDK_ASSEMBLY_NAME` and treating it differently than any other JSII lib. And I also meant having different kinds of namespaces (`NamespaceKind`) just for the purpose of capturing types inside that namespace differently.

I'll take a look at your last change so we can continue this discussion, sorry this review is taking a while.